### PR TITLE
chore: logging

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

Adds logging when a contact is added or updated.

### How to test the changes in this Pull Request:

1. Switch Newspack Plugin to `feat/logger-tweaks` branch (https://github.com/Automattic/newspack-plugin/pull/1807)
1. Set env. variable `NEWSPACK_LOG_LEVEL` to `2`
2. Trigger a contact add (e.g. via the new signup block)
3. Observe an event logged in the debug log (`wp-content/debug.log` if `WP_DEBUG_LOG` is `true`)

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/class-newspack-newsletters-logger.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	public static function log( $payload ) {
+		if ( ! defined( 'NEWSPACK_LOG_LEVEL' ) || 0 > (int) NEWSPACK_LOG_LEVEL ) {
+			return;
+		}
+
+		$header = 'NEWSPACK-NEWSLETTERS';
+		if ( class_exists( '\Newspack\Logger' ) ) {
+			\Newspack\Logger::log( $payload, $header );
+		} else {
+			error_log( '[' . $header . ']: ' . $message ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
```
Comment (miguelpeixe) [inline includes/class-newspack-newsletters-logger.php:28]: `$message` is not defined

**Review by miguelpeixe — CHANGES_REQUESTED** [review]

File: includes/class-newspack-newsletters-logger.php (hunk 1 above)
Comment (adekbadek) [reply includes/class-newspack-newsletters-logger.php]: Fixed in 4ec4724835f7394d2e69b9e011021971704f3090

File: includes/class-newspack-newsletters-logger.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	public static function log( $payload ) {
+		if ( ! defined( 'NEWSPACK_LOG_LEVEL' ) || 0 > (int) NEWSPACK_LOG_LEVEL ) {
+			return;
+		}
+
+		$header = 'NEWSPACK-NEWSLETTERS';
+		if ( class_exists( '\Newspack\Logger' ) ) {
+			\Newspack\Logger::log( $payload, $header );
+		} else {
+			error_log( '[' . $header . ']: ' . $payload ); // phpcs:ignore WordPress.PHP.DevelopmentFunctions.error_log_error_log
```
Comment (miguelpeixe) [inline includes/class-newspack-newsletters-logger.php:28]: If `$payload` is of `any` type, we should cast it to string before `error_log()`.

**Review by miguelpeixe — CHANGES_REQUESTED** [review]

File: includes/class-newspack-newsletters-logger.php (hunk 2 above)
Comment (adekbadek) [reply includes/class-newspack-newsletters-logger.php:28]: Added type checking in 9b07f9c3ddcbbdbc76c48d0ebf559a9f21a8076f. To be extra safe, it'll exit if something else than a string is to be logged.

**Review by miguelpeixe — APPROVED** [review]
To be merged after https://github.com/Automattic/newspack-plugin/pull/1807

**Pull request merged**